app/gui2/src/components/GraphEditor/clipboard.ts

```diff
@@ -120,12 +123,14 @@ function getClipboard() {
 export function useGraphEditorClipboard(
   graphStore: GraphStore,
   selected: ToValue<Set<NodeId>>,
-  createNodes: NodeCreation['createNodes'],
+  createNodes: (nodesOptions: Iterable<NodeCreationOptions>) => void,
 ) {
   /** Copy the content of the selected node to the clipboard. */
   function copySelectionToClipboard() {
     const nodes = new Array<Node>()
-    for (const id of toValue(selected)) {
+    const edit = graphStore.startEdit()
```

Comment:
My understanding was that starting editing is cheap, but applying changes is expensive. Unfortunately, we don’t have any documentation where it will be explained.

Removed the usage of the mutable module here in favor of `syncModule` inside of `pickInCodeOrder`.

Reply:
It does have some cost--it has to copy the state of the module.